server/src/main/java/org/opensearch/index/search/stats/SearchStats.java

```diff
@@ -281,6 +321,12 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeVLong(queryTimeInMillis);
             out.writeVLong(queryCurrent);
 
+            if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
I think this is not a hard requirement, but we usually serialize / deserliaze new fields at the end - it minimizes the impact of  any possible serialization issues.

Reply:
Makes sense! Since I wanted concurrentQueryCount to come right after queryCount in the results I just implemented it like this in all the difference places but that doesn't affect serialization so I'll go ahead and make the change to be safe.